clifford/_multivector.py

```diff
@@ -118,12 +118,42 @@ def vee(self, other) -> 'MultiVector':
         functions instead, as these work in degenerate metrics like PGA too,
         and are equivalent but faster in other metrics.
         """
-        return self.layout.MultiVector(value=self.layout.vee_func(self.value, other.value))
+        other, mv = self._checkOther(other, coerce=True)
+
+        if mv:
+            newValue = self.layout.vee_func(self.value, other.value)
+        else:
+            if isinstance(other, np.ndarray):
+                obj = self.__array__()
+                return obj&other
+
+            newValue = self.value&other
```

Comment:
This line is wrong, it performs bitwise and. This code-path is taken by `(5*e1) & 3`, and should give the same result as if `3` were first turned into a real multivector.

Reply:
Surely in that case 3 would have been coerced into a `MultiVector` during `checkother(3, coerce=True)`?

Reply:
I missed the coerce=True. I'm not certain this line is reachable - so I'd be inclined to raise `TypeError`.

It's pretty obvious that our coercion rules needs some work. I'll maybe revisit this in another PR.